### ISSUE TITLE
Merge proposals instead of overwriting

### ISSUE THREAD

**Issue by kpudlik** [Constellation-Labs/constellation#1073]
![image](https://user-images.githubusercontent.com/9077743/79447866-d0297080-7fe0-11ea-80a6-7c910ccb1072.png)

If we block communication between nodes for a while (so proposal requests are timing out) then it gets no proposals from others which makes fallback to empty Map. Because all nodes timed out and "returned" empty Map it sets `0` as majority for a moment. It is incorrect situation as we should never overwrite proposal  at height. Once set proposal should never change for specific height.

So basically modify `RedownloadService.updatePeerProposal` to set only these heights in `RedownloadService.peersProposals` which haven't been set yet. `peerProposals` should be immutable for specific heights no matter what.
